lib/controller/BibliotheekController.class.php

```diff
@@ -85,333 +134,273 @@ public function wenslijst() {
 	 *
 	 */
 	protected function catalogustonen() {
-		$this->view = new BibliotheekCatalogusContent();
+		$this->view = new BibliotheekCatalogusDatatable();
 	}
 
 	/**
 	 * Inhoud voor tabel op de cataloguspagina ophalen
 	 */
 	protected function catalogusdata() {
-		$catalogus = new BiebCatalogus();
-		$this->view = new BibliotheekCatalogusDatatableContent($catalogus);
+		/**
+		 * @var Boek[] $data
+		 */
+		$data = $this->model->find()->fetchAll();
+		$uid = filter_input(INPUT_GET, "eigenaar", FILTER_SANITIZE_STRING);
+		$results = [];
+		if ($uid !== null) {
+			foreach ($data as $boek) {
+				if ($boek->isEigenaar($uid)) {
+					$results[] = $boek;
+				}
+			}
+		} else {
+			$results = $data;
+		}
+		$this->view = new BibliotheekCatalogusDatatableResponse($results);
 		$this->view->view();
 		exit;
 	}
 
 	/**
-	 * Laad een boek object
-	 *
-	 * ga er van uit dat in getParam(3) een boekid staat en laad dat in.
-	 * @param $boekid $boekid
-	 *          of leeg: gebruikt getParam()
+	 * Boek weergeven
+	 * @param Boek $boek
 	 */
-	private function loadBoek($boekid = null) {
-		if ($this->hasParam(3) OR $boekid !== null) {
-			if ($boekid === null) {
-				$boekid = $this->getParam(3);
+	protected function boek(Boek $boek) {
+		$boekForm = new BoekFormulier($boek);
+		$alleRecensies = $boek->getRecensies();
+		$andereRecensies = [];
+		$mijnRecensie = new BoekRecensie();
+		$mijnRecensie->boek_id = $boek->id;
+		$exemplaarFormulieren = [];
+		foreach ($boek->getExemplaren() as $exemplaar) {
+			if ($exemplaar->magBewerken()) {
+				$exemplaarFormulieren[$exemplaar->id] = new BoekExemplaarFormulier($exemplaar);
 			}
-			if ($this->hasParam(4) AND in_array($this->action, array('bewerkbeschrijving', 'verwijderbeschrijving'))) {
-				$beschrijvingsid = (int)$this->getParam(4);
-			} else {
-				$beschrijvingsid = 0; //nieuwe beschrijving
+		}
+		foreach ($alleRecensies as $recensie) {
+			if ($recensie->schrijver_uid == LoginModel::getUid()) {
+				$mijnRecensie = $recensie;
 			}
-			try {
-				$this->boek = new BewerkBoek($boekid, $beschrijvingsid);
-			} catch (CsrException $e) {
-				setMelding($e->getMessage(), -1);
-				redirect('/bibliotheek/');
+			$andereRecensies[] = $recensie;
+
+		}
+		$recensieForm = new RecensieFormulier($mijnRecensie);
+		if ($boekForm->validate()) {
+			if (!$boek->magBewerken()) {
+				throw new CsrToegangException("U mag dit boek niet bewerken");
```

Comment:
Dit denk ik verder naar boven trekken? Je weet al of je een boek mag bewerken voordat je de form valideert.